### PR TITLE
Assignment5

### PR DESCRIPTION
**Pull request by AaronSauther**

### CODE REVIEW

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs [hunk 1]
```diff
@@ -1,25 +1,186 @@
...
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            Factory.Dispose();
+        }
+
+        [TestMethod]
+        public async Task Get_AuthorsInDB_ReturnsAllAuthors()
```
Comment (mmwoodfo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs]:
```suggestion
        public async Task Get_AuthorsInDB_ReturnsAllAuthors()
```
Naming

File: SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+
+namespace SecretSanta.Business.Dto
+{
+    public class UserInput
+    {
+        [Required]
```
Comment (mmwoodfo) [inline SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs]: I don't think these all need to be required

File: SecretSanta/src/SecretSanta.Business/Dto/Gift.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace SecretSanta.Business.Dto
+{
+    public class Gift : GiftInput, IEntity
+    {
+        public int Id { get; set; }
+
+        public override bool Equals(object? obj)
```
[the remark this answers is not in the document]
Comment (AaronSauther) [reply SecretSanta/src/SecretSanta.Business/Dto/Gift.cs]: Yeah, I used it in some of the unit tests for cleaner code

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/BaseApiControllerTests.cs [hunk 4]
```diff
@@ -11,13 +13,17 @@
 namespace SecretSanta.Api.Tests.Controllers
 {
     [TestClass]
-    public abstract class BaseApiControllerTests<TEntity, TService> 
+    public abstract class BaseApiControllerTests<TEntity, TDto, TInputDto, TService>
```
[the remark this answers is not in the document]
Comment (AaronSauther) [reply SecretSanta/test/SecretSanta.Api.Tests/Controllers/BaseApiControllerTests.cs]: Oh really? I ended up using it in a couple of different tests

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 5]
```diff
@@ -17,7 +18,11 @@ public class Startup
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public static void ConfigureServices(IServiceCollection services)
-        {
+        {
+            services.AddDbContext<ApplicationDbContext>(options =>
+               options.EnableSensitiveDataLogging()
+                      .UseSqlite("Data Source=Blog.db"));
```
[the remark this answers is not in the document]
Comment (AaronSauther) [reply SecretSanta/src/SecretSanta.Api/Startup.cs]: Thanks for the catch, will fix!

File: SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs (hunk 2 above)
Comment (AaronSauther) [reply SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs]: Has been fixed, thank-you!

File: SecretSanta/src/SecretSanta.Web/Startup.cs [hunk 6]
```diff
@@ -12,9 +13,11 @@ public class Startup
     {
         // This method gets called by the runtime. Use this method to add services to the container.
         // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
-        //public void ConfigureServices(IServiceCollection services)
-        //{
-        //}
+        public void ConfigureServices(IServiceCollection services)
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Web/Startup.cs]: Member can be made static

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs [hunk 7]
```diff
@@ -1,25 +1,271 @@
...
+            Assert.IsTrue(DTosAreEqual(expectedDto, returnedDto));
+            Assert.IsTrue(DTosAreEqual(expectedDto, databseDto));
+        }
+
+        [TestMethod]
+        public async Task Delete_ValidID_Success()
+        {
+            //Arrange
+            using ApplicationDbContext dbContext = Factory.GetDbContext();
+            Data.Gift giftToBeDelete = await dbContext.Gifts.FindAsync(1);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllTests.cs]: Local variable 'giftToBeDelete' should be used

**Pull request closed without merging**